server/server.go

```diff
@@ -177,6 +178,7 @@ func (h *handler) handle(pkt *protocol.Packet) {
 	}
 	h.tokens <- struct{}{}
 	h.mtx.Lock()
+	h.conn.SetWriteDeadline(time.Now().Add(h.timeout))
```

Comment:
This returns an error that should be handled and the mutex should be unlocked and the connection should be closed.

Reply:
Seems like https://github.com/cloudflare/gokeyless/blob/efe77714f3928a83da416f15058252466ed2add9/conn/conn.go#L266-L270 handles the error as well